app.py

```diff
@@ -436,8 +436,34 @@ def api_generate():
     if system:
         payload["system"] = system
     
+    # Convert string values to appropriate types for Ollama API
     if options:
-        payload["options"] = options
+        processed_options = {}
+        for key, value in options.items():
+            if key in ['num_ctx', 'num_predict', 'num_keep', 'seed', 'top_k']:
+                # Convert to integer
+                try:
+                    processed_options[key] = int(value) if value != '' and value is not None else None
+                except (ValueError, TypeError):
+                    continue
+            elif key in ['temperature', 'top_p', 'repeat_penalty', 'typical_p']:
+                # Convert to float
+                try:
+                    processed_options[key] = float(value) if value != '' and value is not None else None
+                except (ValueError, TypeError):
+                    continue
+            elif key in ['repeat_last_n']:
```

Comment:
It's good practice to convert to `int` inside a `try-except` block to gracefully handle cases where the value might be empty or not a valid integer. However, the current implementation in `app.py` already handles this. This is a good defensive programming practice.